Mathlib/ModelTheory/Basic.lean

```diff
@@ -240,6 +250,12 @@ theorem card_mk₂ (c f₁ f₂ : Type u) (r₁ r₂ : Type v) :
           Cardinal.lift.{u} #r₁ + Cardinal.lift.{u} #r₂ := by
   simp [card_eq_card_functions_add_card_relations, add_assoc]
 
+instance {f : ℕ → Type*} {R : ℕ → Type*} (n : ℕ) [DecidableEq (f n)] :
+    DecidableEq ((⟨f, R⟩ : Language).Functions n) := inferInstance
+
+instance {f : ℕ → Type*} {R : ℕ → Type*} (n : ℕ) [DecidableEq (R n)] :
+    DecidableEq ((⟨f, R⟩ : Language).Relations n) := inferInstance
```

Comment:
If `inferInstance` can find these, they're probably not very useful as `instances`; so maybe just make them `example`s?

Reply:
These are the things that replace the deleted lines in `ModelTheory.Ring.Basic` - somehow these declarations still help it find these instances. I'd love to know why and find a better solution, but for now, something like these is necessary.

Reply:
Please add a comment to that effect (and an `example` to replace the deleted lines in `ModelTheory.Ring.Basic` to demonstrate that the instances work)